x-pack/plugins/session_view/public/components/DetailPanelCopy/index.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { ReactNode } from 'react';
+import { EuiButtonIcon, EuiCopy } from '@elastic/eui';
+import { DetailPanelListItem } from '../DetailPanelListItem';
+import { dataOrDash } from '../../utils/data_or_dash';
+import { useStyles } from './styles';
+
+interface DetailPanelCopyDeps {
+  children: ReactNode;
+  textToCopy: string | number;
+  display?: 'inlineBlock' | 'block' | undefined;
+}
+
+interface DetailPanelListItemProps {
+  copy: ReactNode;
+  display?: string;
+}
+
+/**
+ * Copy to clipboard component in Session view detail panel.
+ */
+export const DetailPanelCopy = ({
+  children,
+  textToCopy,
+  display = 'inlineBlock',
+}: DetailPanelCopyDeps) => {
+  const styles = useStyles();
+
+  const props: DetailPanelListItemProps = {
+    copy: (
+      <EuiCopy textToCopy={dataOrDash(textToCopy).toString()} display={display}>
+        {(copy) => (
+          <EuiButtonIcon
+            css={styles.copyButton}
+            onClick={copy}
+            display="base"
+            iconType="copyClipboard"
+            size="xs"
+            aria-label="copy"
```

Comment:
To follow i18n standards we translate all of our strings within Kibana. You can look here as an example: https://github.com/elastic/kibana/blob/a29c81b7dbb07b7fc4cba08018adfe0efd740df4/x-pack/plugins/security_solution/public/resolver/view/graph_controls.tsx#L267. Or just search i18n within kibana to see a slew of other examples